This is taken from alpine aports commit:

>   commit 71c7249a3c476549e4214dd5e5caf810f521456b
>   Author: Stefan Krah <skrah@bytereef.org>
>   Date:   Wed Feb 10 17:13:13 2021 +0000
>
>   main/python3: build --with-system-libmpdec

Without this patch, use of --with-system-libmpdec silently does nothing,
and 'python3 -c "import _decimal" will fail.

It should result in
   /usr/lib/python3.*/lib-dynload/_decimal.cpython-3*-x86_64-linux-gnu.so

that has a library dependency on libmpdec like it does on python-3.12.

    $ ldd lib-dynload/_decimal.cpython-312-x86_64-linux-gnu.so
        linux-vdso.so.1 (0x00007ffeacfb2000)
        libmpdec.so.4 => /usr/lib/libmpdec.so.4 (0x00007337c35c2000)
        libc.so.6 => /lib/libc.so.6 (0x00007337c338b000)
        libm.so.6 => /lib/libm.so.6 (0x00007337c32ae000)
        /lib64/ld-linux-x86-64.so.2 (0x00007337c3620000)

diff -Nurp a-Python-3.8.7/Modules/_decimal/_decimal.c b-Python-3.8.7/Modules/_decimal/_decimal.c
--- a-Python-3.8.7/Modules/_decimal/_decimal.c	2020-12-21 16:25:24.000000000 +0000
+++ b-Python-3.8.7/Modules/_decimal/_decimal.c	2021-02-10 15:45:09.976876631 +0000
@@ -3284,7 +3284,7 @@ dec_format(PyObject *dec, PyObject *args
     }
     else {
         size_t n = strlen(spec.dot);
-        if (n > 1 || (n == 1 && !isascii((uchar)spec.dot[0]))) {
+        if (n > 1 || (n == 1 && !isascii((unsigned char)spec.dot[0]))) {
             /* fix locale dependent non-ascii characters */
             dot = dotsep_as_utf8(spec.dot);
             if (dot == NULL) {
@@ -3293,7 +3293,7 @@ dec_format(PyObject *dec, PyObject *args
             spec.dot = PyBytes_AS_STRING(dot);
         }
         n = strlen(spec.sep);
-        if (n > 1 || (n == 1 && !isascii((uchar)spec.sep[0]))) {
+        if (n > 1 || (n == 1 && !isascii((unsigned char)spec.sep[0]))) {
             /* fix locale dependent non-ascii characters */
             sep = dotsep_as_utf8(spec.sep);
             if (sep == NULL) {
diff -Nurp a-Python-3.8.7/setup.py b-Python-3.8.7/setup.py
--- a-Python-3.8.7/setup.py	2020-12-21 16:25:24.000000000 +0000
+++ b-Python-3.8.7/setup.py	2021-02-10 15:45:09.977876640 +0000
@@ -2077,7 +2077,7 @@ class PyBuildExt(build_ext):
         undef_macros = []
         if '--with-system-libmpdec' in sysconfig.get_config_var("CONFIG_ARGS"):
             include_dirs = []
-            libraries = [':libmpdec.so.2']
+            libraries = ['mpdec']
             sources = ['_decimal/_decimal.c']
             depends = ['_decimal/docstrings.h']
         else:
